include/boost/geometry/algorithms/detail/disjoint/point_geometry.hpp

```diff
@@ -39,20 +39,25 @@ namespace detail { namespace disjoint
 {
 
 
-template<typename Point, typename Geometry>
-struct disjoint_point_linear
+struct disjoint_point_segment
 {
-    static inline
-    bool apply(Point const& pt, Geometry const& g)
+    template <typename Point, typename Segment>
+    static inline bool apply(Point const& point, Segment const& segment)
     {
-        return !geometry::covered_by(pt, g);
+        typedef geometry::model::referring_segment<Point const> other_segment;
+
+        other_segment other(point, point);
+        return disjoint_segment
+            <
+                Segment, other_segment
+            >::apply(segment, other);
```

Comment:
What was wrong with the previous version?
Wasn't it the same as `reverse_covered_by` used below for Pt/Linear?
Or is `covered_by()` not defined for Pt/Seg?

Isn't Pt/Seg algorithm faster than Seg/Seg. AFAIU in Seg/Seg points must be checked for equality and then Pt/Seg version called. So there is additional condition and the algorithm contains unnecessary code (Seg/Seg even if only Pt/Seg is needed) which can potentialy harm the instructions cache. Or am I missing something?